### PR TITLE
Fix this e2e failure causes subsequent e2e failures altogether

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/109616/pull-kubernetes-e2e-gce-cos-alpha-features/1607060890820022272/

Failure of this one test will result in failure of all subsequent tests

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/storage/non_graceful_node_shutdown.go [hunk 1]
```diff
@@ -100,8 +100,20 @@ var _ = utils.SIGDescribe("[Feature:NodeOutOfServiceVolumeDetach] [Disruptive] [
...
+					c()
+				}
+				cleanupItems = nil
+			}
+			defer cleanup()
+
 			ginkgo.By("Stopping the kubelet non gracefully for pod" + pod.Name)
 			utils.KubeletCommand(ctx, utils.KStop, c, pod)
+			cleanupItems = append(cleanupItems, func() {
+				utils.KubeletCommand(ctx, utils.KStart, c, pod)
```
Comment (aojea) [inline test/e2e/storage/non_graceful_node_shutdown.go:115]: you can use `ginkgo.DeferCleanup` directly and avoid the `cleanup func()`
cc: @pohly

Comment (pohly) [reply test/e2e/storage/non_graceful_node_shutdown.go:106]: This is using the `ctx` from the test, which is wrong: after a failure, that context is cancelled and cannot be used anymore. A simpler and correct invocation is:

```suggestion
				ginkgo.DeferCleanup(utils.KubeletCommand, utils.KStart, c, pod)
```

Comment (wzshiming) [reply test/e2e/storage/non_graceful_node_shutdown.go]: Done

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Pull request merged**